### DISCUSSION TITLE
Remove Grid Lines from Scanned Images

### DISCUSSION THREAD

**Discussion by sbind29893** [ImageMagick/ImageMagick#2937] [Help]
Hi Everyone,

I need help to remove gridlines from scanned images without hampering the handwritten text.

Pls, find the below sample image.

Thank you in advance.
![Application_Form](https://user-images.githubusercontent.com/33458318/101382799-6139b880-38de-11eb-8fdb-1dfe1eadca8a.png)

**Comment by fmw42** [comment]
I do not think that is possible with ImageMagick. You can remove the grid lines with morphology in ImageMagick, but it will leave gaps in your letters if they intersect with your grid lines.

**Reply by sbind29893** [reply]
Thanks for your reply. What will be the exact command to remove the grid lines?

**Comment by fmw42** [comment]
```
convert form.png -background white -deskew 40% -write mpr:img \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x30 -negate \) \
-evaluate-sequence add \
result.png

```

![result](https://user-images.githubusercontent.com/13458218/101390699-15542880-3878-11eb-8a18-8f14ca4ee9e6.png)

Or do the morphology twice:

```
convert form.png -background white -deskew 40% -write mpr:img \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x30 -negate \) \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x30 -negate \) \
-evaluate-sequence add \
result2.png
```

![result2](https://user-images.githubusercontent.com/13458218/101398320-bfd14900-3882-11eb-82db-23285276a4cf.png)

**Reply by sbind29893** [reply]
Thank you soo much @fmw42 for the help.

**Reply by sbind29893** [reply]
hi @fmw42 

Can you help me to remove gridlines from below cropped images?

![LAABD00035777135_nameValue](https://user-images.githubusercontent.com/33458318/101630804-623b2900-3a49-11eb-89d3-1244d513f7de.png)
![LAABD00035777496_nameValue](https://user-images.githubusercontent.com/33458318/101630808-636c5600-3a49-11eb-90e7-577f7b2ae7c9.png)
![LAABD00035778223_nameValue](https://user-images.githubusercontent.com/33458318/101630809-6404ec80-3a49-11eb-9d14-e4822b9649cf.png)
![LAABD00035778272_nameValue](https://user-images.githubusercontent.com/33458318/101630811-649d8300-3a49-11eb-9957-c6598cf271db.png)
![LAABD00035778384_nameValue](https://user-images.githubusercontent.com/33458318/101630813-649d8300-3a49-11eb-82ff-94da5ee1f999.png)
![LAABD00035781516_nameValue](https://user-images.githubusercontent.com/33458318/101630815-65361980-3a49-11eb-90ce-b1da48e340d4.png)
![LAABD00035782133_nameValue](https://user-images.githubusercontent.com/33458318/101630816-65ceb000-3a49-11eb-8e10-0fdb8ba4919f.png)
![LAABD00035782694_nameValue](https://user-images.githubusercontent.com/33458318/101630817-65ceb000-3a49-11eb-80f9-2316e46c59b0.png)
![LAABD00035782843_nameValue](https://user-images.githubusercontent.com/33458318/101630818-66674680-3a49-11eb-977d-1b4db0e1fc54.png)
![LAABD00035782939_nameValue](https://user-images.githubusercontent.com/33458318/101630821-66ffdd00-3a49-11eb-85bc-5039b498b80b.png)
![LAABD00035783457_nameValue](https://user-images.githubusercontent.com/33458318/101630822-66ffdd00-3a49-11eb-89e9-559bb24c10da.png)
![LAABD00035784380_nameValue](https://user-images.githubusercontent.com/33458318/101630823-67987380-3a49-11eb-96fb-9bd63534f392.png)
![LAABD00035785200_nameValue](https://user-images.githubusercontent.com/33458318/101630825-68310a00-3a49-11eb-9ffc-965f39f18e56.png)
![LAABD00035785546_nameValue](https://user-images.githubusercontent.com/33458318/101630828-68310a00-3a49-11eb-8177-05d40be2e978.png)
![LAABD00035786098_nameValue](https://user-images.githubusercontent.com/33458318/101630830-68c9a080-3a49-11eb-9312-d3cb13697cf1.png)
![LAABD00035786260_nameValue](https://user-images.githubusercontent.com/33458318/101630832-69623700-3a49-11eb-962d-55dded78a7ba.png)
![LAABD00035788489_nameValue](https://user-images.githubusercontent.com/33458318/101630834-69623700-3a49-11eb-8ecd-b1f32c7ac509.png)
![LAABD00035789249_nameValue](https://user-images.githubusercontent.com/33458318/101630837-69facd80-3a49-11eb-8a4b-e1f89d0b45e1.png)
![LAABD00035790665_nameValue](https://user-images.githubusercontent.com/33458318/101630838-6a936400-3a49-11eb-984b-41c6f83732af.png)
![LAABD00035791115_nameValue](https://user-images.githubusercontent.com/33458318/101630840-6a936400-3a49-11eb-8930-00c6fd40b069.png)
![LAABD00035793463_nameValue](https://user-images.githubusercontent.com/33458318/101630841-6b2bfa80-3a49-11eb-97c9-24367375c28d.png)
![LAABD00035793540_nameValue](https://user-images.githubusercontent.com/33458318/101630845-6bc49100-3a49-11eb-850d-a247b076981c.png)
![LAABD00035793673_nameValue](https://user-images.githubusercontent.com/33458318/101630849-6c5d2780-3a49-11eb-90e4-9737d8072ba8.png)
![LAABD00035794559_nameValue](https://user-images.githubusercontent.com/33458318/101630851-6c5d2780-3a49-11eb-9010-f50f000f2755.png)
![LAABD00035794563_nameValue](https://user-images.githubusercontent.com/33458318/101630853-6cf5be00-3a49-11eb-9082-84893b6b5123.png)

**Reply by fmw42** [reply]
For each image do the same as before but since the heights are smaller, you will need to reduce the height in the -morphology but keep it larger than than the grid size.  This may not work as well with such small heights. 

![form](https://user-images.githubusercontent.com/13458218/101660395-fb435300-39fb-11eb-83f8-af7b0797d99a.png)

```
convert form.png -background white -deskew 40% -write mpr:img \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x15 -negate \) \
\( mpr:img -morphology close rectangle:30x1 -negate \) \
\( mpr:img -morphology close rectangle:1x15 -negate \) \
-evaluate-sequence add \
result4.png

```
![result4](https://user-images.githubusercontent.com/13458218/101660424-026a6100-39fc-11eb-89dc-82f7e813dcaa.png)


**Why are you using such small strips? You would be better removing the grid lines on your original full size forms and then cut them into strips if that is what you need. **
